elastic-test/src/main/java/com/hazelcast/jet/tests/elastic/ElasticTest.java

```diff
@@ -0,0 +1,203 @@
+                                .id(Integer.toString(t))
+                                .source(asMap(indexCounter, t)))
+                .bulkRequestFn(() -> new BulkRequest().setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE))
+                .build();
+
+        Pipeline toElastic = Pipeline.create();
+        toElastic.readFrom(TestSources.items(inputItems))
+                .rebalance()
+                .writeTo(elasticSink);
+
+        JobConfig jobConfig = new JobConfig();
+        jobConfig.setName("ElasticTest_writeTo_" + indexCounter);
+        client.newJob(toElastic, jobConfig).join();
+    }
+
+    private void executeReadFromElasticPipeline(JetInstance client, int indexCounter) {
+        final String ip = elasticIp;
+        BatchSource<String> elasticSource = new ElasticSourceBuilder<>()
+                .clientFn(() -> RestClient.builder(new HttpHost(ip, 9200, "http")))
+                .searchRequestFn(() -> new SearchRequest("elastictest-index" + indexCounter))
+                .mapToItemFn(hit -> {
+                    Map<String, Object> sourceAsMap = hit.getSourceAsMap();
+                    return sourceAsMap.get("id") + "_" + sourceAsMap.get("record");
+                })
+                .enableSlicing()
+                .build();
+
+        Pipeline fromElastic = Pipeline.create();
+        fromElastic.readFrom(elasticSource)
+                .writeTo(Sinks.list(SINK_LIST_NAME));
+
+        JobConfig jobConfig = new JobConfig();
+        jobConfig.setName("ElasticTest_readFrom_" + indexCounter);
+        client.newJob(fromElastic, jobConfig).join();
+    }
+
+    private void executeDeleteFromElasticPipeline(JetInstance client, int indexCounter) {
+        final String ip = elasticIp;
+        Sink<Integer> elasticSink = ElasticSinks.builder()
+                .clientFn(() -> RestClient.builder(new HttpHost(ip, 9200, "http")))
+                .<Integer>mapToRequestFn(
+                        t -> new DeleteRequest("elastictest-index" + indexCounter, Integer.toString(t)))
+                .bulkRequestFn(() -> new BulkRequest().setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE))
+                .build();
+
+        Pipeline toElastic = Pipeline.create();
+        toElastic.readFrom(TestSources.items(inputItems))
+                .rebalance()
+                .writeTo(elasticSink);
+
+        JobConfig jobConfig = new JobConfig();
+        jobConfig.setName("ElasticTest_deleteFrom_" + indexCounter);
+        client.newJob(toElastic, jobConfig).join();
+    }
+
+    private void assertResults(JetInstance client, int indexCounter) {
+        IList<String> list = client.getList(SINK_LIST_NAME);
+        assertEquals(itemCount, list.size());
+        List<String> copiedList = new ArrayList<>(list);
+        Collections.sort(copiedList);
+        List<String> expectedList = prepareExpectedList(indexCounter);
+        assertEquals(expectedList, copiedList);
```

Comment:
```suggestion
        IList<String> list = client.getList(SINK_LIST_NAME);
        Set<String> set = new HashSet<>();
        String prefix = "_index-" + indexCounter + "_";
        for(String item: list) {
            assertTrue(item.startsWith(prefix));
            set.add(item);
        }
        assertEquals(itemCount, set.size());
```
This assertion is very costly: we are sorting the IList then create a test list which creates a lot of Strings for each assertion. We don't need to check explicitly for each item I think. asserting that there is no missing or duplicate item and that each item starts with the mentioned index counter should be enough.

Reply:
we can get rid off `prepareExpectedList` if we apply this